eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -0,0 +1,281 @@
+steps:
+- script: |
+    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]%USERPROFILE%\.m2\repository"
+  condition: and(eq(variables['Agent.OS'], 'Windows_NT'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['EnableCache'], 'true'))
+  displayName: Detecting Maven repository on Windows
+- script: |
+    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]$HOME/.m2/repository"
+  condition: and(ne(variables['Agent.OS'], 'Windows_NT'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['EnableCache'], 'true'))
+  displayName: Detecting Maven repository on Linux and macOS
+- task: CacheBeta@0
+  inputs:
+    key: |
+      JobName-$(Agent.JobName)
+      CacheSalt-$(CacheSalt)
+      EnableJetty-$(EnableJetty)
+      $(Build.SourcesDirectory)/pom.client.xml
+      $(Build.SourcesDirectory)/pom.data.xml
+      $(Build.SourcesDirectory)/pom.management.xml
+      $(Build.SourcesDirectory)/pom.xml
+      $(Build.SourcesDirectory)/advisor/resource-manager/v2017_04_19/pom.xml
```

Comment:
@mitchdenny - key segments can now contain wildcards. This should help reduce the number of file paths you need to list here. For example:

`key: JobName-$(Agent.JobName) | CacheSalt-$(CacheSalt) | EnableJetty-$(EnableJetty) | **/pom*.xml`

Reply:
Thanks Will. I have another PR where I have this all setup and it is working. It does improve perf although: